docs/why.rst

```diff
@@ -0,0 +1,111 @@
+functionality of the module being sorted.
+
+See the :doc:`User Guide <guide>` for details.
+
+
+Design Goals
+------------
+
+µsort prefers the simplest code necessary, with predictable and understandable logic,
+to sort imports while maintaining safety of the output with minimal directives from
+developers writing code:
+
+- Simple dataclasses and sorting methods, optimizing for readable and maintainable
+  code rather than absolute performance.
+
+- A single primary code path, to reduce the amount of code and number of test cases
+  needed to guarantee safety.
+
+- Stable sorting methods that should always result in the same final output, even
+  after multiple passes. This is enforced as part of the functional test suite.
+
+- `LibCST`_ for parsing files and modifying syntax trees.
+  µsort transforms CST nodes into dataclasses, then back to CST nodes after sorting.
+  This guarantees grammatical correctness of both the input file and the generated
+  output. Support for future grammar or syntax changes is the responsibility of LibCST.
+
+- No support for configuring output style. µsort works best when run before a
+  dedicated code formatter like `Black`_ for enforcing style choices. µsort does use
+  the configured line length for Black when rendering imports on one or more lines.
+
+- No vendored code. All dependencies are satisfied from PyPI, with liberal version
+  constraints whenever possible.
+
+
+Non-Goals
+---------
+
+There are some features of alternative import sorters that µsort will not offer:
+
+- Broad behavioral changes when sorting files, like moving imports from below a function
+  to the top of the file during development. Guaranteeing safety and stability while
+  also moving imports between blocks is difficult and hard to reason about in code.
+
+- Removing unused imports. This would require µsort to also consume and understand the
+  functionality of each file to accurately know which imports are unused, which would
+  increase the amount of code and testing needed to guarantee safety. Furthemore, the
+  possibility for import-time side effects in removed imports makes this operation
+  potentially dangerous in subtle ways.
+
+- Configurable output style. More options means more code paths to follow, and more
+  test cases needed to cover all possible outputs. This is better left for dedicated
+  code formatters like `Black`_.
+
+- Python 2 support. µsort's use of strict parsing and manipulation of syntax trees
+  makes supporting Python 2 grammar elements a non-starter.
+
+
+Comparison to isort
+-------------------
+
+Unlike `isort`_, µsort only sorts imports within individual "blocks" of imports,
+inferred based on a number of heuristics (see :ref:`Import Blocks <import-blocks>`
+for details). µsort will not attempt to move imports outside of these blocks, or
+across any blocks of code that aren't import statements.
+
+For example, µsort detects two distinct blocks of imports in the following code::
```

Comment:
Might also be worth calling out the human intent of something more like the isort readme's example

```
import a
print("after a")
import b
```

...as people commonly do in debug w/ prints.